components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/config/EmailTemplateManagerImpl.java

```diff
@@ -0,0 +1,358 @@
+import org.wso2.carbon.registry.core.Collection;
+import org.wso2.carbon.registry.core.RegistryConstants;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Provides functionality to manage email templates used in notification emails.
+ */
+public class EmailTemplateManagerImpl implements EmailTemplateManager {
+    private I18nMgtDataHolder dataHolder = I18nMgtDataHolder.getInstance();
+    private RegistryResourceMgtService resourceMgtService = dataHolder.getRegistryResourceMgtService();
+
+    private static final String TEMPLATE_BASE_PATH = I18nMgtConstants.EMAIL_TEMPLATE_PATH;
+    private static final String PATH_SEPARATOR = RegistryConstants.PATH_SEPARATOR;
+    private static final Log log = LogFactory.getLog(EmailTemplateManagerImpl.class);
+
+    private static final String TEMPLATE_REGEX_KEY = I18nMgtConstants.class.getName() + "_" + I18nMgtConstants
+            .EMAIL_TEMPLATE_NAME;
+    private static final String REGISTRY_INVALID_CHARS = I18nMgtConstants.class.getName() + "_" + "registryInvalidChar";
+
+    static {
+        IdentityValidationUtil.addPattern(TEMPLATE_REGEX_KEY, I18nMgtConstants.EMAIL_TEMPLATE_TYPE_REGEX);
+        IdentityValidationUtil.addPattern(REGISTRY_INVALID_CHARS, IdentityValidationUtil.ValidatorPattern
+                .REGISTRY_INVALID_CHARS_EXISTS.getRegex());
+    }
+
+    @Override
+    public void addEmailTemplateType(String emailTemplateDisplayName, String tenantDomain) throws
+            I18nEmailMgtException {
+
+        validateEmailTemplateType(emailTemplateDisplayName, tenantDomain);
+
+        // get template directory name from display name.
+        String normalizedTemplateName = I18nEmailUtil.getNormalizedName(emailTemplateDisplayName);
+
+        // persist the template type to registry ie. create a directory.
+        String path = TEMPLATE_BASE_PATH + PATH_SEPARATOR + normalizedTemplateName;
+        try {
+            // check whether a template exists with the same name.
+            if (resourceMgtService.isResourceExists(path, tenantDomain)) {
+                String errorMsg = String.format(I18nMgtConstants.ErrorMsg.DUPLICATE_TEMPLATE_TYPE,
+                        emailTemplateDisplayName, tenantDomain);
+                throw new I18nEmailMgtClientException(errorMsg);
+            }
+
+            Collection collection = I18nEmailUtil.createTemplateType(normalizedTemplateName, emailTemplateDisplayName);
+            resourceMgtService.putIdentityResource(collection, path, tenantDomain);
+        } catch (IdentityRuntimeException ex) {
+            String errorMsg = String.format("Error adding template type %s to %s tenant.", emailTemplateDisplayName,
+                    tenantDomain);
+            log.error(errorMsg);
+            throw new I18nEmailMgtServerException(errorMsg, ex);
```

Comment:
Either log or throw the error. Not necessary to do both in non-service level classes.